autoload/textobj/lastinserted.vim

```diff
@@ -1,42 +1,47 @@
 let s:pos_insert_enter = [0, 0, 0, 0]
 
 function! textobj#lastinserted#select_i()
-    return s:select('v')
+    return s:select('i')
 endfunction
 
 function! textobj#lastinserted#select_a()
-    return s:select('V')
+    return s:select('a')
 endfunction
 
 function! s:invalid_buf()
     return s:pos_insert_enter[0] != getpos('.')[0]
 endfunction
 
-function! s:invalid_region(first, last)
-    if first[1] > last[1]
-        return 1
-    elseif first[1] == last[1]
-        if first[2] >= last[2]
-            return 1
-        endif
-    endif
-    return 0
+function! s:char_under_cursor()
+    return matchstr(getline('.'), '\%' . col('.') . 'c.')
 endfunction
 
-function! s:select(motion)
-
+function! s:select(mode)
```

Comment:
Why did you change `motion` param name to `mode`? In Vim, mode generally means Vim's mode such as normal mode, insert mode, visual mode and so on.  So we should avoid using the name for other stuffs.